ok , need to review/fixup for failing now test which seems to consider some tricky case...:

```
__________________ test_tabular_summary_avoid_repeated_clear ___________________

    def test_tabular_summary_avoid_repeated_clear():
        out = Tabular(style={"name": {"aggregate": len}})
        out(OrderedDict([("name", "foo")]))
        out(OrderedDict([("name", "bar"),
                         ("new", "col")]))
        # The new column requires a rewrite, but the summary has already been
        # removed, so that shouldn't lead to an additional move-up and clear.
        lines = out.stdout.splitlines()
>       assert_eq_repr(lines[-3],  # -1: summary, -2: bar
                       unicode_cap("cuu1") + unicode_cap("ed") +
                       unicode_cap("cuu1") + "foo    ")

pyout/tests/test_tabular.py:1774: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

a = '\x1b[Afoo    ', b = '\x1b[A\x1b[J\x1b[Afoo    '

    def assert_eq_repr(a, b):
        """Compare the repr's of `a` and `b` to escape escape codes.
        """
>       assert repr(a) == repr(b)
E       assert "'\\x1b[Afoo    '" == "'\\x1b[A\\x1...x1b[Afoo    '"
E         
E         - '\x1b[A\x1b[J\x1b[Afoo    '
E         + '\x1b[Afoo    '

pyout/tests/utils.py:34: AssertionError
```

--- a/pyout/common.py
+++ b/pyout/common.py
@@ -861,7 +861,7 @@ class Content(object):
           * append: the only change in the content is the addition of a line,
             and the returned content will consist of just this line.
 
-          * an integer, N: the Nth line of the output needs to be update, and
+          * an integer, N: the Nth line of the output needs to be updated, and
             the returned content will consist of just this line.
 
           * repaint: all lines need to be updated, and the returned content

--- a/pyout/interface.py
+++ b/pyout/interface.py
@@ -358,17 +358,18 @@ class Writer(object):
             except UnknownColumns as exc:
                 self._columns.extend(exc.unknown_columns)
                 self._init_prewrite()
-                self._write_fn(row, style, redo=True)
+                self._write_fn(row, style)
 
     def _get_last_summary_length(self):
         last_summary = self._last_summary
         return len(last_summary.splitlines()) if last_summary else 0
 
-    def _write_update(self, row, style=None, redo=False):
+    def _write_update(self, row, style=None):
         last_summary_len = self._get_last_summary_length()
+
         content, status, summary = self._content.update(row, style)
 
-        if last_summary_len > 0 and not redo:
+        if last_summary_len > 0:
             # Clear the summary because 1) it has very likely changed, 2)
             # it makes the counting for row updates simpler, 3) and it is
             # possible for the summary lines to shrink.
@@ -410,7 +411,7 @@ class Writer(object):
         self._last_content_len = len(self._content)
         self._last_summary = summary
 
-    def _write_incremental(self, row, style=None, redo=False):
+    def _write_incremental(self, row, style=None):
         content, status, summary = self._content.update(row, style)
         if isinstance(status, int):
             lgr.debug("Duplicating line %d with %r", status, row)
@@ -419,7 +420,7 @@ class Writer(object):
         self._stream.write(content)
         self._last_summary = summary
 
-    def _write_final(self, row, style=None, redo=False):
+    def _write_final(self, row, style=None):
         _, _, summary = self._content.update(row, style)
         self._last_summary = summary
 

